src/Microsoft.Health.Dicom.Core/Features/CustomTag/ICustomTagStore.cs

```diff
@@ -44,5 +47,15 @@ public interface ICustomTagStore
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>The watermark of latest instance.</returns>
         Task<long?> GetLatestInstanceAsync(CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Get instances with watermark upto maxWatermark.
+        /// </summary>
+        /// <param name="maxWatermark">The max watermark</param>
+        /// <param name="top">Top X instanances.</param>
+        /// <param name="indexStatus">The index status.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The instances.</returns>
+        Task<IEnumerable<VersionedInstanceIdentifier>> GetInstancesInThePastAsync(long maxWatermark, int top, IndexStatus indexStatus = IndexStatus.Created, CancellationToken cancellationToken = default);
```

Comment:
This does not fit in CustomtagStore, it is accessing the IndexStore, you should probably move it there

Reply:
Also, any reason to make IndexStatus a param, do we want to pull anything other than created?

Reply:
> This does not fit in CustomtagStore, it is accessing the IndexStore, you should probably move it there

I'm thinking if also move GetLatestIntanceAsync into IndexStore




> Also, any reason to make IndexStatus a param, do we want to pull anything other than created?

I don't think we need IndexStatus rather than Created for now, adding it as param is just for extensibility.